packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,345 @@
+    bytes32 public identifier;
+    SkinnyOptimisticOracleInterface public skinnyOptimisticOracle;
+    OracleAncillaryInterface public oracle;
+    StoreInterface public store;
+
+    struct Transaction {
+        address to;
+        uint256 value;
+        bytes data;
+        Enum.Operation operation;
+    }
+
+    struct Proposal {
+        Transaction[] transactions;
+        uint256 requestTime;
+        bytes explanation;
+    }
+
+    mapping(uint256 => bytes32) public proposalHashes;
+    uint256 public prevProposalId;
+
+    /**
+     * @notice Construct Optimistic Oracle Module.
+     * @param _finder Finder address.
+     * @param _owner Address of the owner.
+     * @param _collateral Address of the ERC20 collateral used for bonds.
+     * @param _bond Bond required (must be at least as large as final fee for collateral type).
+     * @param _rules Reference to the rules for the Gnosis Safe (e.g., IPFS hash or URI).
+     * @param _identifier The approved identifier to be used with the contract, usually "ZODIAC".
+     * @param _liveness The period, in seconds, in which a proposal can be disputed.
+     */
+    constructor(
+        address _finder,
+        address _owner,
+        address _collateral,
+        uint256 _bond,
+        string memory _rules,
+        bytes32 _identifier,
+        uint64 _liveness
+    ) {
+        bytes memory initializeParams = abi.encode(_finder, _owner, _collateral, _bond, _rules, _identifier, _liveness);
+        setUp(initializeParams);
+    }
+
+    function setUp(bytes memory initializeParams) public override initializer {
+        __Ownable_init();
+        (
+            address _finder,
+            address _owner,
+            address _collateral,
+            uint256 _bond,
+            string memory _rules,
+            bytes32 _identifier,
+            uint64 _liveness
+        ) = abi.decode(initializeParams, (address, address, address, uint256, string, bytes32, uint64));
+        finder = FinderInterface(_finder);
+        require(_getCollateralWhitelist().isOnWhitelist(address(_collateral)), "bond token not supported");
+        collateral = IERC20(_collateral);
+        bond = _bond;
+        rules = _rules;
+        require(_getIdentifierWhitelist().isIdentifierSupported(_identifier), "identifier not supported");
+        identifier = _identifier;
+        require(_liveness > 0, "liveness can't be 0");
+        liveness = _liveness;
+        setAvatar(_owner);
+        setTarget(_owner);
+        transferOwnership(_owner);
+        _sync();
+
+        emit OptimisticOracleModuleDeployed(_owner, avatar, target);
+    }
+
+    function priceProposed(
+        bytes32 _identifier,
+        uint32 _timestamp,
+        bytes memory _ancillaryData,
+        SkinnyOptimisticOracleInterface.Request memory _request
+    ) external {
+        emit PriceProposed(_identifier, _timestamp, _ancillaryData, _request);
+    }
```

Comment:
Huh, that looks like a stub whose functionality moved elsewhere and should have been deleted. Getting rid of it.

Reply:
Hmm, removing it breaks some tests. Will investigate further.

Reply:
Ahhh, I remember now. This is a callback function since this contract is an `OptimisticRequester`. It fires off the event with the ancillary data from the Optimistic Oracle after a price is proposed. Will add it back in.